File: AED2/tps/tp1/SomaDigitos.java
import java.util.Scanner;

public class SomaDigitos {

    public static int somaDigitos(int numero) {
        if (numero < 0) {
            numero = -numero;
        }

        if (numero < 10) {
            return numero;
        }

        return numero % 10 + somaDigitos(numero / 10);
    }

    private static int paraInteiro(String texto) {
        if (texto.isEmpty()) {
            return 0;
        }

        int sinal = 1;
        int inicio = 0;

        if (texto.charAt(0) == '-') {
            sinal = -1;
            inicio = 1;
        }

        if (inicio == texto.length()) {
            return 0;
        }

        int numero = 0;

        for (int i = inicio; i < texto.length(); i++) {
            char c = texto.charAt(i);

            if (c < '0' || c > '9') {
                return 0;
            }

            numero = numero * 10 + (c - '0');
        }

        return numero * sinal;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (scanner.hasNextLine()) {
            String linha = scanner.nextLine();
            System.out.println(somaDigitos(paraInteiro(linha)));
        }

        scanner.close();
    }
}

File: AED2/tps/tp1/somaDigitos.c
```c
#include <stdio.h>

int somaDigitos(int numero) {
    int soma = 0;

    if (numero < 0) {
        numero = -numero;
    }

    while (numero > 0) {
        soma += numero % 10;
        numero /= 10;
    }

    return soma;
}

int paraInteiro(const char *texto) {
    int numero = 0;
    int sinal = 1;
    int i = 0;

    if (texto[0] == '-') {
        sinal = -1;
        i = 1;
    }

    while (texto[i] != '\0') {
        if (texto[i] < '0' || texto[i] > '9') {
            return 0;
        }

        numero = numero * 10 + (texto[i] - '0');
        i++;
    }

    return numero * sinal;
}

int main(void) {
    char linha[105];

    while (fgets(linha, sizeof(linha), stdin) != NULL) {
        int i = 0;

        while (linha[i] != '\0' && linha[i] != '\n' && linha[i] != '\r') {
            i++;
        }

        linha[i] = '\0';

        printf("%d\n", somaDigitos(paraInteiro(linha)));
    }

    return 0;
}
```

